### ISSUE TITLE
uBlock Origin v1.14.4: No more updates of filter lists when addon "RequestPolicy Continued" uses default policy "block all"

### ISSUE THREAD

**Issue by potassiumchloride** [gorhill/uBlock#2965]
Dear gorhill,

after upgrading uBlock Origin from legacy version 1.13.8 to web extension version 1.14.4 I can no longer receive updates of my filter lists (also when I manually force them) and it always tells me "A network error prevented the resource from being updated".

After some tests I found out that the update mechanism of 1.14.4 for the filter lists conflicts with my installed addon "RequestPolicy Continued" when in that addon the default policy "Block requests by default" has been chosen. Updates of filter lists work perfectly in uBlock Origin 1.13.8 with this block-all setting in RequestPolicy Continued. Alternatively, if I change the default policy in RequestPolicy Continued from "Block requests by default" to "Allow requests by default" the filter lists in 1.14.14 receives updates again. However, this work-around is not really desirable as I prefer the "black-list all, white-list selectively" approach.

Short summary:

Browser Firefox 55.0.3, 64bit
- uBlock Origin 1.13.8 with RequestPolicy Continued  "Block requests by default" --> works
- uBlock Origin 1.14.4 with RequestPolicy Continued  "Block requests by default" --> no updates of filter lists
- uBlock Origin 1.14.4 with RequestPolicy Continued  "Allow requests by default" --> works, but not desirable

Currently, I have downgraded from 1.14.4 to 1.13.8 until a solution can be found. At this point I would also like to take the chance to thank you for your tremendous work on uBlock Origin (and uMatrix!) which are really amazing and great addons!

**Comment by Cyrixo** [comment]
Hello,
I have the same problem.
I don't have RequestPolicy installed.
If I disable uMatrix, the updates work

**Comment by gorhill** [comment]
> uBlock Origin 1.14.4 with RequestPolicy Continued "Block requests by default" --> no updates of filter lists
> uBlock Origin 1.14.4 with RequestPolicy Continued "Allow requests by default" --> works, but not desirable

So RequestPolicy Continued ("RPC") is blocking uBO's own requests... Hence not a uBO issue?

Adjust your RPC rules or report to the maintainers of RPC.

**Comment by potassiumchloride** [comment]
Dear gorhill,

thanks for your quick response. I don't understand why you closed this issue so fast by assuming that RPC is the guilty player. I use uBO and RPC together since many, many months and there has never been any kind of problem in the interaction between both of them. I clearly explained that uBO 1.14.4 fails to update filter lists while this doesn't happen with uBO <= 1.13.8 whereas the RPC version is absolutely unchanged at that time. If I can provide any further information to help or find out the reason, please let me know.

**Comment by gorhill** [comment]
It's all in the release notes, uBO is now webext. The main difference as far as RPC is concerned -- a guess -- is that uBO's URLs are now `moz-extension://` instead of `chrome://`. So the issue must be simply a rule issue in RPC.

**Comment by potassiumchloride** [comment]
Hi gorhill,

many thanks for your further investigation, explanations and clarification (also for your comment in the RPC issue tracker at https://github.com/RequestPolicyContinued/requestpolicy/issues/844). I added a white-list entry at the RPC settings for destination moz-extension://*:* and installed uBO 1.14.4 again but it still gives the same network error for updating the filter lists. I'm going to wait and see if the RPC developer will also investigate this issue. Thanks again for your quick responses and help!

**Comment by gorhill** [comment]
> I added a white-list entry at the RPC settings for destination `moz-extension://*:*` and installed uBO 1.14.4 again but it still gives the same network error for updating the filter lists.

Should `moz-extension://*:*` be used for the source instead?

uBO, the source, `moz-extension://` is fetching something from outside, let's say at `https://example.com`.

**Comment by gorhill** [comment]
By the way, just out of curiosity, have you tried dynamic filtering in advanced mode in uBO?

**Comment by potassiumchloride** [comment]
Hi gorhill,

amazing, you did it! Your final suggestion to use `moz-extension://*:*` as source instead of destination in RPC solved the whole problem! Many thanks to you for your help in tracking down the issue and finally discovering a solution, especially since it turned out that the problem and solution was not to be found in your addon but in another developer's addon. Surely, this hint will also be helpful to other people having similar problems with RPC and uBO. Lots of thanks for your effort and work!

**Comment by potassiumchloride** [comment]
With respect to your question about my usage of dynamic filtering in uBO: Since the first day and although I have activated advanced mode in uBO I have never really tried to use the dynamic filtering because I found RPC was more intuitive in handling. However, on another Firefox machine I use uMatrix (uM) (instead of RPC and NoScript) to handle 3rd party requests. If RPC doesn't get converted to webext for Firefox 57 I'll completely switch to uBO and uM to manage these things and then I'll try to get more familiar with the dynamic filtering mode in uBO.
